From cfcd8613dcf57678458fd6e6a9adc8304229d19b Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 23 Jul 2014 15:01:44 +0100
Subject: [PATCH 081/118] sparsify: Relax requirement that output device cannot
 be block device (RHBZ#1122557).

To fix RHBZ#1056290, I prevented virt-sparsify being used if the
output device is a block device.

I have now retested this scenario and it does work (in both copying
and in-place mode), and does not delete the output device, and
therefore we can relax this restriction so only char devices are
banned.

This is useful for oVirt which uses a qcow2 formatted block device to
store virtual machines.

(cherry picked from commit 12b54ef5e3aecd55d228a96dfb0a9ab5c8acdf4a)
---
 sparsify/sparsify.ml | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/sparsify/sparsify.ml b/sparsify/sparsify.ml
index d01899b..d6b1193 100644
--- a/sparsify/sparsify.ml
+++ b/sparsify/sparsify.ml
@@ -171,11 +171,7 @@ read the man page virt-sparsify(1).
   if contains_colon outdisk then
     error (f_"output filename '%s' contains a colon (':'); qemu-img command line syntax prevents us from using such an image") outdisk;
 
-  (* Check the output is not a block or char special (RHBZ#1056290). *)
-  if is_block_device outdisk then
-    error (f_"output '%s' cannot be a block device, it must be a regular file")
-      outdisk;
-
+  (* Check the output is not a char special (RHBZ#1056290). *)
   if is_char_device outdisk then
     error (f_"output '%s' cannot be a character device, it must be a regular file")
       outdisk;
-- 
2.7.4

